src/WingetCreateCLI/Program.cs

```diff
@@ -1,88 +1,104 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license.
-
-namespace Microsoft.WingetCreateCLI
-{
-    using System;
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.WingetCreateCLI
+{
+    using System;
     using System.Linq;
-    using System.Threading.Tasks;
-    using CommandLine;
+    using System.Threading.Tasks;
+    using CommandLine;
     using CommandLine.Text;
-    using Microsoft.WingetCreateCLI.Commands;
-    using Microsoft.WingetCreateCLI.Logging;
-    using Microsoft.WingetCreateCLI.Properties;
-    using Microsoft.WingetCreateCLI.Telemetry;
-    using Microsoft.WingetCreateCLI.Telemetry.Events;
+    using Microsoft.WingetCreateCLI.Commands;
+    using Microsoft.WingetCreateCLI.Logging;
+    using Microsoft.WingetCreateCLI.Properties;
+    using Microsoft.WingetCreateCLI.Telemetry;
+    using Microsoft.WingetCreateCLI.Telemetry.Events;
     using Microsoft.WingetCreateCore.Common;
-
-    /// <summary>
-    /// Main entry class for the CLI.
-    /// </summary>
-    internal class Program
+
+    /// <summary>
+    /// Main entry class for the CLI.
+    /// </summary>
+    internal class Program
     {
-        private static async Task<int> Main(string[] args)
+        private static async Task<int> Main(string[] args)
         {
             Logger.Initialize();
             UserSettings.FirstRunTelemetryConsent();
             TelemetryEventListener.EventListener.IsTelemetryEnabled();
-
-            string arguments = string.Join(' ', Environment.GetCommandLineArgs());
-            Logger.Trace($"Command line args: {arguments}");
-
-            Parser myParser = new Parser(config => config.HelpWriter = null);
+
+            string arguments = string.Join(' ', Environment.GetCommandLineArgs());
+            Logger.Trace($"Command line args: {arguments}");
+
+            Parser myParser = new Parser(config => config.HelpWriter = null);
 
             var types = new Type[] { typeof(NewCommand), typeof(UpdateCommand), typeof(SubmitCommand), typeof(SettingsCommand), typeof(TokenCommand), typeof(CacheCommand) };
             var parserResult = myParser.ParseArguments(args, types);
 
             BaseCommand command = parserResult.MapResult(c => c as BaseCommand, err => null);
 
-            if (command == null)
+            string token = command?.GitHubToken ?? GitHubOAuth.ReadTokenCache();

```

Comment:
Why not just call command.LoadGitHubClient() and then use that in GetLatestRelease instead of passing in a token?

Reply:
I like this idea, but it will require some changes like making LoadGitHubClient() a public method as well as storing the token as a private static string in the GitHub.cs class in order to keep GetLatestRelease static.